Mathlib/Topology/Algebra/ConstMulAction.lean

```diff
@@ -286,6 +297,46 @@ theorem interior_smul (c : G) (s : Set α) : interior (c • s) = c • interior
 #align interior_smul interior_smul
 #align interior_vadd interior_vadd
 
+section GroupSeparation
+
+variable [T2Space α]
+
+/--
+If the action of a group `G` on `α` is continuous, then for all points not fixed by `g : G`,
+there exists an open set `s` such that `x ∈ s` and `g • s` is disjoint from `s`.
+-/
+@[to_additive "If the action of a group `G` on `α` is continuous, then for all points not fixed
+by `g : G`, there exists an open set `s` such that `x ∈ s` and `g +ᵥ s` is disjoint from `s`."]
+theorem t2_separation_smul {x : α} {g : G} (gx_ne : g • x ≠ x) :
+    ∃ s : Set α, IsOpen s ∧ x ∈ s ∧ Disjoint s (g • s) :=
+  let ⟨s, t, s_open, t_open, gx_in_s, x_in_t, disj_st⟩ := t2_separation gx_ne
+  ⟨g⁻¹ • s ∩ t,
+    (s_open.smul g⁻¹).inter t_open,
+    ⟨Set.mem_inv_smul_set_iff.mpr gx_in_s, x_in_t⟩,
+    by
+      rw [Set.smul_set_inter, smul_inv_smul]
+      exact (disj_st.symm.inter_left' _).inter_right _⟩
```

Comment:
Can you make this proof shorter by proving it after the result below?

Reply:
Definitely, yes. Originally I had a very complex proof that proved the `Set.Finite` version using `t2_separation_smul`

Reply:
It unfortunately makes the proof longer, because `PairwiseDisjoint` and `InjOn` require me to jump through more hoops.